content/en/blog/2020/custom-istio-dns-name/index.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "Customizing Istio DNS stub names"
+subtitle: Changing or adding stub domains to make ServiceEntry hostnames routable
+description: Customizing global or region hostnames as declared in ServiceEntry(s) for routability.
+publishdate: 2020-05-05
+attribution: "Christian Posta (Solo.io)"
+keywords: [istio, dns, mTLS, coredns]
+---
+
+Istio can be run across multiple clusters and has the necessary configuration APIs to support [shared control plane](/docs/setup/install/multicluster/shared/) or a non-shared, [replicated control model](/docs/setup/install/multicluster/gateways). The motivation for deploying across multiple clusters has been covered excellently by fellow Istio contributor [Vadim Eisenberg](https://twitter.com/vadimeisenberg) in the [Multi-Mesh Deployments for Isolation and Boundary Protection](/blog/2019/isolated-clusters/) blog.
+
+One aspect of multi-cluster functionality enabling services to communicate with each other is service discovery. In an Istio replicated control-plane multi-cluster setup, you create [ServiceEntry](/docs/reference/config/networking/service-entry/)s in one cluster to point to services running in another cluster. For example:
+
+    {{< text bash >}}
+    $ kubectl apply --context=$CTX_CLUSTER1 -n foo -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: httpbin-bar
+    spec:
+      hosts:
+      # must be of form name.namespace.global
+      - httpbin.bar.global
+      # Treat remote cluster services as part of the service mesh
+      # as all clusters in the service mesh share the same root of trust.
+      location: MESH_INTERNAL
+      ports:
+      - name: http1
+        number: 8000
+        protocol: http
+      resolution: DNS
+      addresses:
+      - 240.0.0.2
+      endpoints:
+      - address: ${CLUSTER2_GW_ADDR}
+        ports:
+          http1: 15443 # Do not change this port value
+    EOF
+    {{< /text >}}
+
+This `ServiceEntry` creates an entry in the Istio service-discovery registry for `httpbin.bar.global` with endpoints that point to a different cluster's gateway. Without any other configuration changes, this hostname, `httpbin.bar.global` is not routable. In the docs for setting up multi-cluster replicated control plane, however, there is a [section for configuring the DNS stubbing](/docs/setup/install/multicluster/gateways/#setup-dns) for your Kubernetes cluster. When we set up the DNS stubbing, we tell `kube-dns` to delegate queries for `.global` to the Istio coredns component.
```

Comment:
```suggestion
This `ServiceEntry` creates an entry for `httpbin.bar.global` in the Istio service-discovery registry. The new entry's endpoints direct traffic to a different cluster's gateway. Without any other configuration changes, this hostname, `httpbin.bar.global`, is not routable. To make it routable, follow the instructions about [configuring the DNS stubbing](/docs/setup/install/multicluster/gateways/#setup-dns) for your Kubernetes cluster. When you set up the DNS stubbing, you tell `kube-dns` to delegate queries for `.global` to the Istio `coredns` component.
```

My proposed revision here is meant to simplify the paragraph, but I may have gone too far and changed the meaning; please ignore or change anything that I got wrong here. My main idea here is to focus on what the reader should do rather than on where the documentation that you're directing them to lives.

(I do like the idea of saying where the docs are, but I think in this particular sentence, it distracts a little bit from the focus of the paragraph. But I may be wrong.)